frontend/src/components/feature/chat/ChatMessage/Renderers/FileMessage.tsx

```diff
@@ -104,6 +115,14 @@ const PDFPreviewButton = ({ message, user }: {
 
     const fileName = getFileName(message.file)
 
+    const image = convertPdfToImage(message, user)
```

Comment:
The image variable is being assigned the result of a Promise chain, not the actual converted image. The convertPdfToImage function's result is not being properly captured or returned. The then block doesn't return anything, so image will be a Promise that resolves to undefined. Also, "image" is not even being used anywhere, so I'm not sure about this implementation. Did it work for you?

Reply:
I was working on implementation, but due to time constraints, I had to create a PR . Nevertheless I'll finish the implementation over this weekend.

Reply:
okay, you can move the PR in draft mode until you are done and then re request a review.